src/ipyradiant/visualization/vis_tool.py

```diff
@@ -0,0 +1,114 @@
+from pathlib import Path
+
+import traitlets as T
+
+import ipywidgets as W
+from ipycytoscape import CytoscapeWidget
+from rdflib import Graph, Literal, URIRef
+from rdflib.namespace import RDF
+
+
+class CytoscapeVisualization(W.GridBox):
+    graph = T.Instance(Graph, allow_none=True)
+    cyto_widget = T.Instance(CytoscapeWidget, allow_none=True)
+    nodes = T.List()
+    click_output = T.Instance(W.Output)
+    click_output_box = T.Instance(W.VBox)
+    selected_node = T.Unicode(allow_none=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.edge_color = kwargs.get("edge_color", "pink")
+        self.node_color = kwargs.get("node_color", "grey")
+        self.output_title = W.HTML("<h3>Output from Clicks</h3>")
+        self.click_output_box.children = [self.output_title, self.click_output]
+
+        self.default_edge = {"selector": "edge", "css": {"line-color": self.edge_color}}
+        self.default_node = {
+            "selector": "node",
+            "css": {"background-color": self.node_color},
+        }
+
+        self.children = [self.cyto_widget, self.click_output_box]
```

Comment:
since the output box is kinda.... ugly... maybe hide it behind a flag? e.g.

```
show_log = T.Boolean(False)
...
T.link((self, 'show_log'), (self.click_output_box.layout, "visible"), lambda x: not x)
```

You may then need to use flex layout instead of grid layout, as it would probably leave empty space.

Reply:
Would I also need to put an observer on show_log then? so that when that variable changes it automatically updates the flexbox?

Reply:
links take care of that: they're like declarative observers.

Reply:
sounds good.. i had to change it to something like: 
`show_outputs = T.Unicode('hidden')`
`T.link((self, 'show_outputs'), (self.click_output_box.layout, 'visibility'))`

because the box doesn't have a visible attribute, will this work also?

Reply:
right, sorry. `T.dlink` is the one that takes a transformer. Always favor _data_ (numbers, indexes in list, booleans) rather than _magic strings_ as the interface between things. so it might actually be:

```py
T.dlink((self, 'show_log'), (self.click_output_box.layout, "visible"), lambda x: "visible" if x else "hidden")
```

Reply:
sounds good. that seems to do the trick!